diff --git a/src/utils.h b/src/utils.h
index a914f21..d42f117 100644
--- a/src/utils.h
+++ b/src/utils.h
@@ -48,22 +48,8 @@
 
 #define xassert(cond, msg) if ((cond) == 0) _err_fatal_simple_core(__func__, msg)
 
-#if defined(__GNUC__) && !defined(__clang__)
-#if defined(__i386__)
-static inline unsigned long long __rdtsc(void)
-{
-    unsigned long long int x;
-    __asm__ volatile (".byte 0x0f, 0x31" : "=A" (x));
-    return x;
-}
-#elif defined(__x86_64__)
-static inline unsigned long long __rdtsc(void)
-{
-    unsigned hi, lo;
-    __asm__ __volatile__ ("rdtsc" : "=a"(lo), "=d"(hi));
-    return ( (unsigned long long)lo)|( ((unsigned long long)hi)<<32 );
-}
-#endif
+#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
+#include <x86intrin.h>
 #endif
 
 typedef struct {
